tests/test_json_excel_compare.py

```diff
@@ -0,0 +1,146 @@
+# This file is part of ts_hvac.
+#
+# Developed for the Vera C. Rubin Observatory Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import asynctest
+import logging
+from lsst.ts.hvac.xml import hvac_mqtt_to_SAL_XML as xml
+
+logging.basicConfig(
+    format="%(asctime)s:%(levelname)s:%(name)s:%(message)s", level=logging.DEBUG
+)
+
+
+class JsonExcelCompareTestCase(asynctest.TestCase):
+    async def setUp(self):
+        """Setup the unit test.
+        """
+        self.log = logging.getLogger("JsonExcelCompareTestCase")
+        self.csv_topics = {}
+        self.json_topics = {}
+
+    def load_csv(self):
+        xml.use_csv = True
+        xml.collect_topics_and_items()
+        self.csv_topics = xml.hvac_topics.copy()
+
+    def load_json(self):
+        xml.use_csv = False
+        xml.collect_topics_and_items()
+        self.json_topics = xml.hvac_topics.copy()
+
+    async def test_compare_csv_and_json(self):
+        self.load_csv()
+        self.load_json()
+        json_topics_missing_in_csv = []
+        csv_topics_missing_in_json = []
+        topic_types_differing = []
+        idl_types_differing = []
+        units_differing = []
+        limits_differing = []
+
+        for json_topic in sorted(self.json_topics.keys()):
+            if json_topic not in sorted(self.csv_topics.keys()):
+                json_topics_missing_in_csv.append(json_topic)
```

Comment:
I strongly urge you to use sets, instead of loops, to find missing keys. It is much more idiomatic and much more efficient.

In fact it turns out that dict keys work as sets for this, so you can skip constructing the sets at all:
```
json_topic_names = self.json_topics.keys()  # or set(self,json_topics.keys()) but you don't need that
csv_topic_names = self.csv_topics.keys()
json_topics_missing_in_csv = json_topic_names - csv_topic_names  # the result is a set
csv_topics_missing_in_json = csv_topic_names - json_topic_names
```
I think the rest of your code will need the loops.

Reply:
This really shows that I am still new to Python programming and still am thinking too much in Java. Thanks!